### PR TITLE
Pvayladev 1013

### PR DESCRIPTION
**Pull request by jansu76**
https://jira.csc.fi/browse/PVAYLADEV-1013

### CODE REVIEW

File: src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java [hunk 1]
```diff
@@ -80,13 +79,18 @@
     private String publicKeyFilePassword;
     private int keyLength;
 
+    public ConsumerGateway() {
+        log.info("!!!!!!!!!!!! ConsumerGateway()");
```
Comment (olli-lindgren) [inline src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java]: Is this a leftover from something?

File: src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java [hunk 2]
```diff
@@ -81,12 +80,13 @@
...
 
     /**
-     * Reads properties (overridden is tests)
+     * Reads properties (overridden in tests)
      * @return
      */
     protected GatewayProperties readGatewayProperties() {
         log.debug("Reading Consumer and ConsumerGateway properties");
         String propertiesDirectory = RESTGatewayUtil.getPropertiesDirectory();
+        Properties testprops = System.getProperties();
```
Comment (jhyoty) [inline src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java]: Unused?

File: src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java [hunk 3]
```diff
@@ -770,19 +791,19 @@ protected void serializeRequest(ServiceRequest request, SOAPElement soapRequest,
                 // Create new symmetric encrypter using of defined key length
                 Encrypter symmetricEncrypter = RESTGatewayUtil.createSymmetricEncrypter(this.keyLength);
                 // Process request parameters
```
Comment (jhyoty) [inline src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java]: the comment is misleading

File: src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java [hunk 4]
```diff
@@ -770,19 +791,19 @@ protected void serializeRequest(ServiceRequest request, SOAPElement soapRequest,
...
-                writeGetParametersToBody(request, payload);
+                writeBodyContents(request, payload);
                 // Process request body
                 if (this.requestBody != null && !this.requestBody.isEmpty()) {
-                    if (convertPost) {
-                        convertJsonToXmlAndWriteToSoap(this.requestBody, soapRequest);
-                    } else {
-                        // send HTTP POST as an attachment
+                    if (!convertPost) {
```
Comment (jhyoty) [inline src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java:800]: Could be combined with the enclosing "if".

File: src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java [hunk 5]
```diff
@@ -681,62 +709,32 @@ public RequestSerializer(String resourceId, String requestBody, String contentTy
...
         @Override
         protected void serializeRequest(ServiceRequest request, SOAPElement soapRequest, SOAPEnvelope envelope) throws SOAPException {
-            writeGetParametersToBody(request, soapRequest);
+            writeBodyContents(request, soapRequest);
             if (this.requestBody != null && !this.requestBody.isEmpty()) {
-                if (convertPost) {
-                    convertJsonToXmlAndWriteToSoap(this.requestBody, soapRequest);
-                } else {
-                    // send HTTP POST as an attachment
+                if (!convertPost) {
```
Comment (jhyoty) [inline src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java:714]: Could be combined with the enclosing "if".

File: src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java (hunk 4 above)
Comment (jansu76) [reply src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java:800]: Retained the separate if-statement for (in my view) better clarity; added else-block and more comments

File: src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java (hunk 5 above)
Comment (jansu76) [reply src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java:714]: Retained the separate if-statement for (in my view) better clarity; added else-block and more comments

**Pull request merged**